bedrock/firefox/views.py

```diff
@@ -118,16 +115,39 @@ def stub_attribution_code(request):
         codes['source'] = 'www.mozilla.org'
         codes['medium'] = '(none)'
 
-    codes['timestamp'] = str(int(time()))
+    response = HttpResponseJSON(sign_attribution_codes(codes))
+    patch_response_headers(response, 300)  # 5 min
+    return response
+
+
+def get_attrribution_code(codes):
+    """
+    Take the attribution codes and return the URL encoded string
+    respecting max length.
+    """
     code = '&'.join('='.join(attr) for attr in codes.items())
+    if len(codes['campaign']) > 5 and len(code) > settings.STUB_ATTRIBUTION_MAX_LEN:
+        # remove 5 char at a time
+        codes['campaign'] = codes['campaign'][:-5] + '_'
+        code = get_attrribution_code(codes)
```

Comment:
If I'm reading this right, if a param other than `utm_campaign` makes this string longer than `STUB_ATTRIBUTION_MAX_LEN`, we allow it? This is likely an edge case, but curious as to the intended handling.

Reply:
Yeah. My thinking is that if we can't seem to fix it just send it on and hope for the best. I didn't see the point in blocking it. Though maybe we should since we know the attribution service will?

Reply:
Updated to return 400 error when code would be > max rather than sending a code that would be too long anyway. We can adjust max via a setting so this should be decently future proof.